File: backend/src/main/java/io/kongkham/kcurr/ApiClients/CurrencyBeacon/CurrencyBeaconExchangeRateApiResponseResponse.java
package io.kongkham.kcurr.ApiClients.CurrencyBeacon;

import lombok.Data;

import java.util.HashMap;

@Data
public class CurrencyBeaconExchangeRateApiResponseResponse {
    private String date;
    private String base;
    private HashMap<String, Double> rates;
}

File: backend/backend/ApiClients/CurrencyBeacon/CurrencyBeaconExchangeRateApiResponseResponse.cs
using System.Text.Json.Serialization;

namespace backend.ApiClients.CurrencyBeacon;

public class CurrencyBeaconExchangeRateApiResponseResponse
{
    [JsonPropertyName("date")]
    public string Date { set; get; }
    [JsonPropertyName("base")]
    public string Base { set; get; }
    [JsonPropertyName("rates")]
    public Dictionary<string, double> Rates { set; get; }
}